### PR TITLE
Add support to set context attributes while indexing an item into cloudsearch.

### PR DESCRIPTION
**Pull request by sarthak-kaps**
Add support to set context attributes while indexing an item into cloudsearch.

### CODE REVIEW

File: indexing/src/main/java/com/google/enterprise/cloudsearch/sdk/indexing/IndexingItemBuilder.java [hunk 1]
```diff
@@ -428,6 +429,20 @@ public IndexingItemBuilder setSearchQualityMetadataQuality(FieldOrValue<Double>
     return this;
   }
 
+  /**
```
Comment (emailofanuj) [inline indexing/src/main/java/com/google/enterprise/cloudsearch/sdk/indexing/IndexingItemBuilder.java:432]: how was this comment added? Was this generated or added manually?

File: indexing/src/main/java/com/google/enterprise/cloudsearch/sdk/indexing/IndexingItemBuilder.java [hunk 2]
```diff
@@ -428,6 +429,20 @@ public IndexingItemBuilder setSearchQualityMetadataQuality(FieldOrValue<Double>
...
+   * Set the {@code contextAttributes} field value for the {@code ItemMetadata}.
+   * <p>
+   *     Using this setter will override the default value for contextAttributes field which is empty.
+   * </p>
+   *
+   * @param contextAttributes the {@code contextAttributes} instance
+   * @return this instance
+   */
+  public IndexingItemBuilder setContextAttributes(List<ContextAttribute> contextAttributes) {
```
Comment (emailofanuj) [inline indexing/src/main/java/com/google/enterprise/cloudsearch/sdk/indexing/IndexingItemBuilder.java:441]: Can we add/modify E2E integration test to make sure that this change works?

**Review by emailofanuj — CHANGES_REQUESTED** [review]

File: indexing/src/main/java/com/google/enterprise/cloudsearch/sdk/indexing/IndexingItemBuilder.java (hunk 1 above)
Comment (sarthak-kaps) [reply indexing/src/main/java/com/google/enterprise/cloudsearch/sdk/indexing/IndexingItemBuilder.java:432]: I have added this comment, similar in style to other comments.

File: indexing/src/main/java/com/google/enterprise/cloudsearch/sdk/indexing/IndexingItemBuilder.java (hunk 2 above)
Comment (sarthak-kaps) [reply indexing/src/main/java/com/google/enterprise/cloudsearch/sdk/indexing/IndexingItemBuilder.java:441]: I have added a basic integration test which indexes 2 items with context attributes and checks if the item is searchable. I have also manually checked that context attributes are getting populated for these items. When I send a manual search request with context attributes I am seeing the feature working as expected.

**Review by emailofanuj — APPROVED** [review]

**Pull request merged**